samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,255 @@
+import boto3
+import re
+import logging
+import jinja2
+from typing import *
+
+LOG = logging.getLogger(__name__)
+
+
+def _extract_api_id_from_arn(api_arn: str) -> str:
+    """
+    Extracts an api id from an HTTP or REST api arn.
+
+    NOTE: https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html
+
+    REST API Arn: `arn:partition:apigateway:region::/restapis/api-id`
+
+    HTTP API Arn: `arn:partition:apigateway:region::/apis/api-id`
+
+    Parameters
+    ----------
+    api_arn : str
+        The arn of the HTTP or REST api from which the api id is extracted.
+
+    Returns
+    -------
+    str
+        The api id from the api arn.
+    """
+
+    return api_arn[api_arn.rindex("/") + 1 :]
```

Comment:
I don't see what input would cause this to fail, since this function is only executed if a string matches 
`^arn:{partition_ex}:apigateway:{region_ex}::\/restapis\/\w+$` 
or 
`^arn:{partition_ex}:apigateway:{region_ex}::\/apis\/\w+$`
meaning it will only extract from valid [HTTP or REST api arns](https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html)

To validate, I'm pretty sure the api id is 10 alphanumeric characters (from this [example](https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html)), but I cannot find it formally documented. I can make the regex more specific to reflect the length of the api id.

Reply:
I see, my concern was around not getting an arn but something else (I am not sure how this parameter is populated).

Reply:
This function is only called on strings that match the expression representing an APIGW RestApi or Api